src/include/duckdb/optimizer/matcher/set_matcher.hpp

```diff
@@ -79,6 +81,18 @@ class SetMatcher {
 				}
 			}
 			return true;
+		} else if (policy == Policy::PARTIAL_ORDERED) {
+			// partial ordered policy, if too many entries are provided, return false
+			if (matchers.size() < entries.size()) {
```

Comment:
I would imagine this should works the other way around - i.e. we need to match all matchers, but if there are extra expressions they do not count. Otherwise the `bindings` provided to a function are off as they may contain fewer entries than expected.

Reply:
I thought that too, but then how do we match the extra expressions when they are provided? Maybe the FunctionExpressionMatcher can have a minimum_matches variable? And the match function has an extra check like 
```c++
if (entries.size() < minimum_matches) {
    return false;
}
```

Reply:
Ah, I think the way the optimizer works changed since my last review - you do want partial matching on the matchers' side. Perhaps call it `SOME_ORDERED` instead of `PARTIAL_ORDERED` for clarity, as you essentially want `SOME` but ordered?

Reply:
yea. For the regex optimiser to work, I need to match 2 or 3 arguments. In both cases the arguments have to be ordered correctly.
I will change the name to `PARTIAL_ORDERED`. Is the `minimum_matches` variable still a good idea?

Reply:
Fine by me - but then it should apply to some as well